contracts/dao/voting/vesting-lp-vault/src/contract.rs

```diff
@@ -0,0 +1,343 @@
+    let new_atom_oracle_contract = deps.api.addr_validate(&new_atom_oracle_contract)?;
+    let new_usdc_oracle_contract = deps.api.addr_validate(&new_usdc_oracle_contract)?;
+    let new_manager = new_manager
+        .map(|new_manager| deps.api.addr_validate(&new_manager))
+        .transpose()?;
+
+    if info.sender != config.owner && new_owner != config.owner {
+        return Err(ContractError::OnlyOwnerCanChangeOwner {});
+    };
+    if info.sender != config.owner && new_vesting_lp_contract != config.vesting_lp_contract {
+        return Err(ContractError::OnlyOwnerCanChangeVestingLpContract {});
+    };
+
+    config.owner = new_owner;
+    config.vesting_lp_contract = new_vesting_lp_contract;
+    config.atom_oracle_contract = new_atom_oracle_contract;
+    config.usdc_oracle_contract = new_usdc_oracle_contract;
+    config.manager = new_manager;
+    config.name = new_name;
+    config.description = new_description;
+    config.validate()?;
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "update_config")
+        .add_attribute("description", config.description)
+        .add_attribute("owner", config.owner)
+        .add_attribute("vesting_lp_contract", config.vesting_lp_contract)
+        .add_attribute("atom_oracle_contract", config.atom_oracle_contract)
+        .add_attribute("usdc_oracle_contract", config.usdc_oracle_contract)
+        .add_attribute(
+            "manager",
+            config
+                .manager
+                .map(|a| a.to_string())
+                .unwrap_or_else(|| "None".to_string()),
+        ))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::VotingPowerAtHeight { address, height } => {
+            to_binary(&query_voting_power_at_height(deps, env, address, height)?)
+        }
+        QueryMsg::TotalPowerAtHeight { height } => {
+            to_binary(&query_total_power_at_height(deps, env, height)?)
+        }
+        QueryMsg::Info {} => query_info(deps),
+        QueryMsg::Dao {} => query_dao(deps),
+        QueryMsg::Name {} => query_name(deps),
+        QueryMsg::Description {} => query_description(deps),
+        QueryMsg::GetConfig {} => query_config(deps),
+        QueryMsg::ListBonders { start_after, limit } => {
+            query_list_bonders(deps, start_after, limit, None)
+        }
+        QueryMsg::BondingStatus { height, address } => {
+            to_binary(&query_bonding_status(deps, env, height, address)?)
+        }
+    }
+}
+
+pub fn query_voting_power_at_height(
+    deps: Deps,
+    env: Env,
+    address: String,
+    height: Option<u64>,
+) -> StdResult<VotingPowerAtHeightResponse> {
+    let config = CONFIG.load(deps.storage)?;
+    let height = height.unwrap_or(env.block.height);
+
+    let query_msg = VestingLpQueryMsg::UnclaimedAmountAtHeight { address, height };
```

Comment:
this query is outdated. please get familiar with the new way of querying vesting contracts at height: https://github.com/neutron-org/neutron-tge-contracts/blob/main/packages/vesting-base/src/msg.rs#L139

there's also a readme: https://github.com/neutron-org/neutron-tge-contracts/tree/main/packages/vesting-base

Reply:
done!